src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -424,10 +487,37 @@ private static void ValidateBuffersList(IList<ArraySegment<byte>> buffers)
         }
 
         /// <summary>Completes the SocketAsyncEventArg's Task with the result of the send or receive, and returns it to the specified pool.</summary>
-        private static void CompleteSendReceive(Int32TaskSocketAsyncEventArgs saea, bool isReceive)
+        private static void CompleteAccept(Socket s, TaskSocketAsyncEventArgs<Socket> saea)
+        {
+            // Pull the relevant state off of the SAEA
+            SocketError error = saea.SocketError;
+            Socket acceptSocket = saea.AcceptSocket;
+
+            // Synchronize with the initiating thread. If the synchronous caller already got what
+            // it needs from the SAEA, then we can return it to the pool now. Otherwise, it'll be
+            // responsible for returning it once it's gotten what it needs from it.
+            bool responsibleForReturningToPool;
+            AsyncTaskMethodBuilder<Socket> builder = saea.GetCompletionResponsibility(out responsibleForReturningToPool);
```

Comment:
I like out vars in ifs, but I personally don't like it used this way, as it makes it hard for me to find / see where the variable is defined.